### ISSUE TITLE
MSSQL incorrectly escapes column names in groupBy operation

### ISSUE THREAD

**Issue by dryu** [yiisoft/yii2#8824]
Yii 2.0.3

```php
$data = (new Query())->select(...)->from(...)->where(...)
->groupBy("DATEPART(year, T.last_update_date), DATEPART(dy, T.last_update_date)")
```

SQL produced:

```
GROUP BY DATEPART(year, [T].[last_update_date)], DATEPART(dy, [T].[last_update_date)] 
```

By definition ```groupBy``` should not escape any column names if it contains some parenthesis (e.g. DB expression).

**Comment by samdark** [comment]
That's expected. You have to use [Expression](https://github.com/yiisoft/yii2/blob/master/framework/db/Expression.php#L17) in non-trivial cases.

**Comment by dryu** [comment]
```php
    /**
     * Sets the GROUP BY part of the query.
     * @param string|array $columns the columns to be grouped by.
     * Columns can be specified in either a string (e.g. "id, name") or an array (e.g. ['id', 'name']).
     * The method will automatically quote the column names unless a column contains some parenthesis
     * (which means the column contains a DB expression).
     * @return static the query object itself
     * @see addGroupBy()
     */
    public function groupBy($columns)
```

```The method will automatically quote the column names unless a column contains some parenthesis```

Either the program does not perform as advertised, or you need to correct the declaration.

Besides, wrapping up the argument to ```groupBy()``` into ```new Expression()``` doesn't help, I can only make it work if manually escaping all columns or splitting the expression in halves and wrapping up each to ```Expression```.

**Comment by cebe** [comment]
@dryu should be fixed by #10682, thanks for reporting!
